core/src/main/java/io/grpc/ServerInterceptors.java

```diff
@@ -80,6 +84,50 @@ public static ServerServiceDefinition intercept(ServerServiceDefinition serviceD
     return serviceDefBuilder.build();
   }
 
+  /**
+   * Create a new {@code ServerServiceDefinition} whose {@link MethodDescriptor} serializes to
+   * and from byte[] for all methods.  The {@code ServerCallHandler} created will automatically
+   * convert back to the original types for request and response before calling the existing
+   * {@code ServerCallHandler}.  Calling this method combined with the intercept methods will
+   * allow the developer to choose whether to intercept messages of byte[], or the modeled types
+   * of their application.
+   *
+   * @param serviceDef the service definition to convert messages to byte[]
+   * @return a wrapped version of {@code serviceDef} with the byte[] conversion applied.
+   */
+  public static ServerServiceDefinition useByteArrayMessages(
+      final ServerServiceDefinition serviceDef) {
+    final MethodDescriptor.Marshaller<byte[]> marshaller =
```

Comment:
Allowing for any generic Marshaller type here and using a ForwardingAndConvertingServerCall would allow for things like never reading the input stream but still returning a cached output stream backed by a ByteBuf if for example the "cache key" came in as a header or something. This isn't possible in the current implementation.  Let me know if you think I should go down this path and actually making a Forwarding* class that is easier maintained as well.